### DISCUSSION TITLE
Mihomo插件 TCP Dial 全球直连 出现大量connection refused

### DISCUSSION THREAD

**Discussion by BarneyTTGumble** [MetaCubeX/mihomo#1726] [General]
### 用Mihomo一段时间了，发现日志出现大量全球直连有connection refused

Mihomo 主要配置如下：
# 用于下载订阅时指定UA
global-ua: clash
    
# 全局配置
port: 7890
socks-port: 7891
redir-port: 7892
mixed-port: 7893
tproxy-port: 7894
ipv6: false
allow-lan: true
unified-delay: true
tcp-concurrent: true
# interface-name: eth0  （路由器下根据情况指定出站接口）
authentication:
  - name:passwd
  
geodata-mode: false
# GEO 文件加载模式（standard：标准加载器/memconservative：专为内存受限 (小内存) 设备优化的加载器 (默认值)）
geodata-loader: standard
geo-auto-update: true
geo-update-interval: 48
geox-url:
  geosite: "https://mirror.ghproxy.com/https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geosite.dat"
  mmdb: "https://mirror.ghproxy.com/https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geoip-lite.metadb"
  geoip: "https://mirror.ghproxy.com/https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/geoip-lite.dat"
  asn: "https://mirror.ghproxy.com/https://github.com/MetaCubeX/meta-rules-dat/releases/download/latest/GeoLite2-ASN.mmdb"
  
# 控制面板
external-controller: 0.0.0.0:9090
secret: ""
external-ui: ui
external-ui-url: "https://mirror.ghproxy.com/https://github.com/MetaCubeX/metacubexd/archive/refs/heads/gh-pages.zip"
# 下载面板地址可更换：https://mirror.ghproxy.com/https://github.com/MetaCubeX/Yacd-meta/archive/gh-pages.zip

# 匹配进程 always/strict/off
find-process-mode: strict
global-client-fingerprint: chrome
keep-alive-idle: 600
keep-alive-interval: 30

# 策略组选择和fakeip缓存
profile:
  store-selected: true
  store-fake-ip: true
  
# 流量嗅探
sniffer:
  enable: true
  sniff:
    HTTP:
      ports: [80, 8080-8880]
      override-destination: true
    TLS:
      ports: [443, 8443]
    QUIC:
      ports: [443, 8443]
  force-domain:
    - +.v2ex.com
  skip-domain:
    - "Mijia Cloud"
    - "dlg.io.mi.com"
    - "+.push.apple.com"
    - "+.apple.com"
    
# 代理模式
tun:
  enable: true
  stack: mixed
  mtu: 9000
  dns-hijack:
    - "any:53"
    - "tcp://any:53"
  auto-route: true
  auto-redirect: true
  auto-detect-interface: true
  
# DNS模块
dns:
  enable: true
  prefer-h3: true
  listen: 0.0.0.0:1053
  ipv6: false
  default-nameserver:
    - 223.5.5.5
    - 119.29.29.29
  respect-rules: true
  # 模式切换 redir-host / fake-ip
  enhanced-mode: fake-ip
  fake-ip-range: 28.0.0.1/8
  # 模式切换 whitelist/blacklist 
  # 黑名单模式表示如果匹配成功则不返回 Fake-IP, 白名单模式时只有匹配成功才返回 Fake-IP
  fake-ip-filter-mode: blacklist
  fake-ip-filter:
    - "+.lan"
    - "+.local"
    - geosite:private
    - geosite:cn
  nameserver:
    - 8.8.8.8
    - 8.8.4.4
    - 1.1.1.1
  proxy-server-nameserver:
    - https://120.53.53.53/dns-query
    - https://223.5.5.5/dns-query
  nameserver-policy:
    "rule-set:private_domain,cn_domain":
      - https://223.5.5.5/dns-query
      - https://120.53.53.53/dns-query
    "rule-set:geolocation-!cn":
      - tls://dns.google
      - tls://1dot1dot1dot1.cloudflare-dns.com
    
   
proxies:
- name: "🟢 直连"
  type: direct
  udp: true

日志如下：
<html>
<body>

序列号 | 类型 | 内容
-- | -- | --
7802 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7801 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7800 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7799 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7798 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7797 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7796 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7795 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7794 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7793 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61074 --> 39.174.79.72:6881 error: dial tcp 39.174.79.72:6881: connect: connection refused
7792 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7791 | [info] | [TCP] 192.168.10.253:61073 --> mtalk.google.com:443 match RuleSet(google_domain) using 🍀 Google[🇺🇸 美国 03 \| 专线]
7790 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7789 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7788 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7787 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7786 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7785 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7784 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7783 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7782 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7781 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7780 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61072 --> 183.250.45.225:6881 error: dial tcp 183.250.45.225:6881: connect: connection refused
7779 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7778 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7777 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7776 | [info] | [TCP] 192.168.10.222:47400 --> szextshort.weixin.qq.com:80 match RuleSet(cn_domain) using 🎯 全球直连[🟢 直连]
7775 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7774 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7773 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7772 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7771 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61071 --> 223.78.209.241:6881 error: dial tcp 223.78.209.241:6881: connect: connection refused
7770 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61070 --> 58.19.95.170:6881 error: dial tcp 58.19.95.170:6881: connect: connection refused
7769 | [warning] | [TCP] dial 🎯 全球直连 (match RuleSet/cn_ip) 192.168.10.253:61070 --> 58.19.95.170:6881 error: dial tcp 58.19.95.170:6881: connect: connection refused

</body>
</html>

**Comment by BarneyTTGumble** [comment]
请问DNS或者哪些配置需要调整，谢谢！DNS这样配置正确吗？

**Comment by morytyann** [comment]
WPS导致的，把192.168.10.253机器上的WPS卸载干净就好了。
